utils/big_decimal.cc

```diff
@@ -143,6 +143,35 @@ std::strong_ordering big_decimal::operator<=>(const big_decimal& other) const
     return x.compare(y) <=> 0;
 }
 
+std::strong_ordering big_decimal::operator<=>(const big_decimal& other) const
+{
+    if (_scale == other._scale) {
+        return _unscaled_value.compare(other._unscaled_value) <=> 0;
+    }
+
+    const static double sqrt_10 = std::sqrt(10);
+
+    const int64_t bit_count = boost::multiprecision::msb(_unscaled_value) - boost::multiprecision::lsb(_unscaled_value);
+    const int64_t bit_count_other = boost::multiprecision::msb(other._unscaled_value) - boost::multiprecision::lsb(other._unscaled_value);
+
+    const int64_t diff_scale_2 = bit_count - bit_count_other;
+    const double diff_scale_2_adj_10 = double(diff_scale_2) / sqrt_10;
+    const double diff_scale_10 = _scale - other._scale;
```

Comment:
Don't understand sqrt(10) here.

I'd expect logarithms. v = unscaled * 10^scale. Then log10 v = log10(unscaled) + scale =~ log2(unscaled)/log2(10) + scale. log2(unscaled) can be approximated with msb(abs(unscaled)), but I don't see sqrt(10) anywhere.

Reply:
Yes, I should have used `log2(10)`, this is fixed in v2.